htdocs/public/partnership/new.php

```diff
@@ -211,17 +217,70 @@ function llxFooterVierge()
-		$partnership->lastname    = GETPOST('lastname');
-		$partnership->address     = GETPOST('address');
-		$partnership->zip         = GETPOST('zipcode');
-		$partnership->town        = GETPOST('town');
-		$partnership->email       = GETPOST('email');
-		$partnership->country_id  = GETPOST('country_id', 'int');
-		$partnership->state_id    = GETPOST('state_id', 'int');
-		//$partnership->typeid      = $conf->global->PARTNERSHIP_NEWFORM_FORCETYPE ? $conf->global->PARTNERSHIP_NEWFORM_FORCETYPE : GETPOST('typeid', 'int');
+		$partnership->status      = 0;
 		$partnership->note_private = GETPOST('note_private');
+		$partnership->date_creation = dol_now();
+		$partnership->date_partnership_start = dol_now();
+		$partnership->fk_user_creat=$user->id;
+
+		$partnershipt->fetch(0, 'default');
+		if ($partnershipt->id > 0) {
+			$partnership->fk_type = $partnershipt->id;
+		}
+
+
+		//$partnership->firstname   = GETPOST('firstname');
+		//$partnership->lastname    = GETPOST('lastname');
+		//$partnership->address     = GETPOST('address');
+		//$partnership->zip         = GETPOST('zipcode');
+		//$partnership->town        = GETPOST('town');
+		//$partnership->email       = GETPOST('email');
+		//$partnership->country_id  = GETPOST('country_id', 'int');
+		//$partnership->state_id    = GETPOST('state_id', 'int');
+		//$partnership->typeid      = $conf->global->PARTNERSHIP_NEWFORM_FORCETYPE ? $conf->global->PARTNERSHIP_NEWFORM_FORCETYPE : GETPOST('typeid', 'int');
+
+		// test if societe already exist
+		$sql =	"SELECT rowid FROM ".MAIN_DB_PREFIX."societe WHERE nom='".$db->escape(GETPOST('societe'))."'";
+		$result = $db->query($sql);
+		if ($result) {
+			$num = $db->num_rows($result);
+		}
+		if ($num = 0) { // si il ya pas d'entree sur le nom  on teste l'email
+			$sql1 =	"SELECT rowid FROM ".MAIN_DB_PREFIX."societe WHERE email='".$db->escape(GETPOST('email'))."'";
+			$result1 = $db->query($sql1);
+			if ($result1) {
+				$num1 = $db->num_rows($result1);
+			}
+			if ($num1 != 0) {
+				$error++;
+				$errmsg = "email already exists please rewrite your company name";
+			} else {
+				//create thirdparty
+				$company = new Societe($db);
+
+				$company->address     = GETPOST('address');
+				$company->zip         = GETPOST('zipcode');
+				$company->town        = GETPOST('town');
+				$company->email       = GETPOST('email');
+				$company->country_id  = GETPOST('country_id', 'int');
+				$company->state_id    = GETPOST('state_id', 'int');
+				$company->name_alias  = dolGetFirstLastname(GETPOST('firstname'), GETPOST('lastname'));
+
+				$resultat=$company->create($user);
+				if ($resultat < 0) {
+					$error++;
+					$errmsg .= join('<br>', $company->errors);
+				}
+
+				$partnership->fk_soc = $company->id;
+			}
+		} elseif ($num > 1) {
+			$error++;
+			$errmsg = 'more than one entry exist for this company please contact us to complete your partnership request';
+		} else {
+			$company = $db->fetch_object($result);
```

Comment:
If we used before the $company->fetch() the the id of company is already into $company->id so the fetch_object can be removed.